lib/assessment/decision_trees.py

```diff
@@ -172,7 +186,61 @@ def u3l18_position_assessment(self, data):
     # No Evidence: No elements placed using the coordinate system.
     else:
       self.assessment = "No Evidence"
+
+  def u3l18_algorithms_conditionals_assessment(self, data):
+
+    conditionals_in_draw_loop = [statement for statement in data["conditionals"] if statement['draw_loop']]
+    
+    user_trigger = False
+    user_triggered_conditionals = [statement for statement in conditionals_in_draw_loop if "user_interaction" in statement.keys() and statement["user_interaction"]]
+    if user_triggered_conditionals:
+      user_trigger = True
+      self.user_trigger_conditional_evidence(user_triggered_conditionals)
+    
+    value_trigger = False
+    variables_in_code = [statement['identifier'] for statement in data['variables']]
+    if variables_in_code:
+      variables_in_conditional_tests = [statement for statement in conditionals_in_draw_loop if 'identifier' in statement.keys() and statement['identifier'] in variables_in_code]
+      for statement in conditionals_in_draw_loop:
+        if 'left' in statement:
+          if type(statement['left']) == dict and 'identifier' in statement["left"].keys():
+            variables_in_conditional_tests.append(statement)
+          elif type(statement['right']) == dict and 'identifier' in statement['right'].keys():
+            variables_in_conditional_tests.append(statement)
+      self.variable_triggered_conditional_evidence(variables_in_conditional_tests)
```

Comment:
This is a little confusing- I think some comments here would help, and since I think it looks like a complicated subset of `conditionals_in_draw_loop`, maybe breaking it out in a helper function and calling `list(filter(helper_function, conditionals_in_draw_loop))`